### PR TITLE
[HOTFIX][MI100][OCL] Remove unsed kernel arg for wrw fp32 NCHW

### PR DESCRIPTION
**Pull request by carlushuang**
This resolve issue https://github.com/ROCmSoftwarePlatform/MIOpen/issues/1012, when under fp32 wrw, it seems feed in more kernel arg than needed. This seems OK under HIP, but in OCL it will report issue. But anyway we need keep consistent kernel arg with metadata declaration.

@shaojiewang please double check this fp32 wrw

### CODE REVIEW

**Review by shaojiewang — DISMISSED** [review]

**Review by atamazov — CHANGES_REQUESTED** [review]
Thanks for quick response. Constness of buffers also needs to be fixed, see https://github.com/ROCmSoftwarePlatform/MIOpen/issues/1012#issuecomment-868895217.

What about regression tests?

**Comment by atamazov** [comment]
> This seems OK under HIP.

I think this is not ok, https://github.com/ROCmSoftwarePlatform/MIOpen/issues/1012#issuecomment-868896151

**Comment by carlushuang** [comment]
> Constness of buffers also needs to be fixed, see #1012 (comment).
OK, let me add that part into this PR

> What about regression tests?
I guess this fix does not introduce regression? If can't work (e.g. after this PR fail on HIP), the CI will just fail

**Comment by atamazov** [comment]
@carlushuang 
> > What about regression tests?
> 
> I guess this fix does not introduce regression? If can't work (e.g. after this PR fail on HIP), the CI will just fail

Or course, not. But this PR should add some regression tests that should prevent this bug in the future (see #1012). If you need assistance, please let me know and I'll add them to this branch.

**Comment by carlushuang** [comment]
@atamazov ok sure, I think need your help on adding some tests in this branch, in case I'm not quite clear of the regression of this PR.
Thanks :)

**Comment by atamazov** [comment]
@carlushuang 
> @atamazov ok sure, I think need your help on adding some tests in this branch, in case I'm not quite clear of the regression of this PR.  Thanks :)

Done in b6a3ff7. I would like to make this commit absolutely clear for you _before_ merging this. Please review this commit and compare it with the description of #1012. Ask if something is unclear.

**Comment by carlushuang** [comment]
@atamazov the flag `--disable-validation` is added, because we don't care about the result is correct or not, but this test it self will fail (because of extra kernel arg) right?

**Comment by atamazov** [comment]
We didn't have a MI100 test for OpenCL backend, so the test weren't run. dca7901 resolves this.

**Review by atamazov — DISMISSED** [review]

**Review by atamazov — APPROVED** [review]

File: test/CMakeLists.txt [hunk 1]
```diff
@@ -1169,3 +1169,24 @@ add_custom_test(test_regression_half_vega FLOAT_DISABLED HALF_ENABLED GFX908_DIS
...
+
+set(ENVS_REGRESSION_ISSUE_1012
+    MIOPEN_DEBUG_IMPLICIT_GEMM_FIND_ALL_SOLUTIONS=1
+    MIOPEN_FIND_MODE=normal)
+
+set(ARGS_REGRESSION_ISSUE_1012
+    --verbose
+    --disable-forward
+    --disable-backward-data
+    --disable-validation)
```
Comment (atamazov) [inline test/CMakeLists.txt:1181]: :anchor:

**Pull request merged**